batch/batch/worker/worker.py

```diff
@@ -2218,7 +2220,10 @@ async def kill(self):
         if self.container is not None:
             await self.container.remove()
 
-    async def execute(self, classpath: str, scratch_dir: str, log_file: str, jar_url: str, argv: List[str]):
+    # TODO This is not backwards compatible
```

Comment:
How hard is it to make a new release with the client changes first that accept an optional batch id argument? If the batch id is not there, then revert back to the old submission method. Otherwise, use the new submission method. I just want to make sure we don't break anyone's code before a release is made. Regardless, we'll need to make a new release ASAP after this merges.

Reply:
We can instead sidestep all of this and do this with no incompatible changes. What do you think about adding an optional parameter to the deploy config that is the batch the job is currently running in? That would work both for JVMJobs and DockerJobs, and no worries about compatibility or reverting

Reply:
Can you point me to where the JVM jobs deploy config is created?

Reply:
Looks like it's created for all jobs here https://github.com/hail-is/hail/blob/7e644d28b53c59d5339073576257f0fb27bdef4f/batch/batch/front_end/front_end.py#L943. The deploy config is currently the same for every job so we just have it stored in the secret `worker-deploy-config`, but it doesn't seem unreasonable to craft a deploy-config that is job-specific.

Reply:
I like this train of thought. Why don't we write a new file like `batch-config` into the job's container disk space where the log file goes already and the new code can read from this file to get the batch id?